src/helics/core/zmq/ZmqCommsSS.cpp

```diff
@@ -0,0 +1,543 @@
+/*
+ Copyright © 2017-2018,
+ Battelle Memorial Institute; Lawrence Livermore National Security, LLC; Alliance for Sustainable Energy, LLC
+ All rights reserved. See LICENSE file and DISCLAIMER for more details.
+ */
+#include "ZmqCommsSS.h"
+#include "../../common/zmqContextManager.h"
+#include "../../common/zmqHelper.h"
+#include "../../common/zmqSocketDescriptor.h"
+#include "../ActionMessage.hpp"
+#include "../NetworkBrokerData.hpp"
+#include "ZmqRequestSets.h"
+//#include <boost/asio.hpp>
+//#include <csignal>
+#include <memory>
+#include <iostream>
+
+static const int DEFAULT_BROKER_PORT_NUMBER = 23414;  // Todo define a different port number
+static const int TX_RX_MSG_COUNT = 20;
+
+using namespace std::chrono;
+/** bind a zmq socket, with a timeout and timeout period*/
+static bool bindzmqSocket (zmq::socket_t &socket,
```

Comment:
Someday (probably not now) we should extract some of these functions that are common between the ZMQ implementations and put them in a separate file to reduce code duplication

Reply:
Ok, leaving it as is for now.